### PR TITLE
added a submission page for edit, new version and deployment forms

### PR DESCRIPTION
**Pull request by ARADDCC002**

### CODE REVIEW

**Review by a3957273 — CHANGES_REQUESTED** [review]

File: pages/model/[uuid]/deploy.tsx [hunk 1]
```diff
@@ -7,12 +7,13 @@ import Paper from '@mui/material/Paper'
...
 import { postEndpoint } from '../../../data/api'
 import { Schema, SplitSchema, Step } from '../../../types/interfaces'
-import { getStepsData, getStepsFromSchema } from '../../../utils/formUtils'
+import { createStep, getStepsData, getStepsFromSchema } from '../../../utils/formUtils'
 import Grid from '@mui/material/Grid'
 import Box from '@mui/material/Box'
 import SchemaSelector from '../../../src/Form/SchemaSelector'
 import SubmissionError from '../../../src/Form/SubmissionError'
 import Form from '../../../src/Form/Form'
+import DeploymentSubmission from '@/src/Form/DeploymentSubmission'
```
Comment (a3957273) [inline pages/model/[uuid]/deploy.tsx:16]: Doesn't follow import of other imports, ../../../

File: pages/model/[uuid]/deploy.tsx [hunk 2]
```diff
@@ -21,6 +22,26 @@ const uiSchema = {
...
+  setActiveStep: Function,
+  onSubmit: Function,
+  _openValidateError: boolean,
+  _setOpenValidateError: Function,
+  _modelUploading: boolean
+) {
+  const data = getStepsData(splitSchema)
+
+  return (
+    <>
```
[the remark this answers is not in the document]
Comment (ARADDCC002) [reply pages/model/[uuid]/deploy.tsx:39]: Resolved

File: pages/model/[uuid]/new-version.tsx [hunk 3]
```diff
@@ -16,6 +16,7 @@ import Form from '../../../src/Form/Form'
 import RenderFileTab, { FileTabComplete } from '../../../src/Form/RenderFileTab'
 import useCacheVariable from '../../../utils/useCacheVariable'
 import LoadingBar from '../../../src/common/LoadingBar'
+import ModelExportAndSubmission from '@/src/Form/ModelExportAndSubmission'
```
[the remark this answers is not in the document]
Comment (ARADDCC002) [reply pages/model/[uuid]/new-version.tsx:19]: Think this was auto import! Resolved

File: pages/model/[uuid]/edit/[version].tsx [hunk 4]
```diff
@@ -8,13 +8,14 @@ import Wrapper from '../../../../src/Wrapper'
...
 import { SplitSchema, Step } from '../../../../types/interfaces'
-import { getStepsData, getStepsFromSchema } from '../../../../utils/formUtils'
+import { createStep, getStepsData, getStepsFromSchema } from '../../../../utils/formUtils'
 
 import SubmissionError from '../../../../src/Form/SubmissionError'
 import Form from '../../../../src/Form/Form'
 import { useGetModelVersion } from 'data/model'
 import { putEndpoint } from 'data/api'
 import useCacheVariable from 'utils/useCacheVariable'
+import ModelEditSubmission from '@/src/Form/ModelEditSubmission'
```
[the remark this answers is not in the document]
Comment (ARADDCC002) [reply pages/model/[uuid]/edit/[version].tsx:18]: Resolved

File: pages/model/[uuid]/new-version.tsx [hunk 5]
```diff
@@ -64,6 +93,24 @@ function Upload() {
...
+        state: {},
+        schemaRef: cModel.schemaRef,
+
+        type: 'Message',
+        index: steps.length,
+        section: 'submission',
+
+        render: () => <></>,
+        renderButtons: renderSubmissionTab,
+        isComplete: () => true,
```
Comment (a3957273) [inline pages/model/[uuid]/new-version.tsx:110]: Similar `isComplete` consideration.

**Review by a3957273 — CHANGES_REQUESTED** [review]

**Review by a3957273 — APPROVED** [review]

**Pull request merged**